### PR TITLE
[20086] Methods to configure transport scenarios

### PR DESCRIPTION
**Pull request by MiguelCompany**
## Description

Open as draft:

- [x] Configure using code
- [x] Configure using environment variable
- [x] Configure using XML
- [x] Tests
- [x] Documentation

@Mergifyio backport 2.12.x 2.11.x 2.10.x 2.6.x

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [x] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- [x] Any new/modified methods have been properly documented using Doxygen. 
- [x] Changes are ABI compatible. 
- [x] Changes are API compatible. 
- [x] New feature has been added to the `versions.md` file (if applicable).
- [x] New feature has been documented/Current behavior is correctly described in the documentation. 
    Related documentation PR: https://github.com/eProsima/Fast-DDS-docs/pull/619
- [x] Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Review by EduPonz — CHANGES_REQUESTED** [review]

File: test/blackbox/common/BlackboxTestsTransportCustom.cpp [hunk 1]
```diff
@@ -305,4 +307,181 @@ TEST(ChainingTransportTests, tcp_client_server_with_wan_correct_sender_resources
...
+{
+    std::vector<std::string> bt_list;
+    bt_list.push_back("participant_none");
+    bt_list.push_back("participant_default");
+    bt_list.push_back("participant_defaultv6");
+    bt_list.push_back("participant_shm");
+    bt_list.push_back("participant_udp");
+    bt_list.push_back("participant_udpv6");
+    bt_list.push_back("participant_largedata");
+    bt_list.push_back("participant_largedatav6");
```
[the remark this answers is not in the document]
Comment (cferreiragonz) [reply test/blackbox/common/BlackboxTestsTransportCustom.cpp:448]: Wrong configuration test has been added in the XML Parser unittest

**Review by EduPonz — DISMISSED** [review]

**Comment by cferreiragonz** [comment]
@richiprosima please test this

**Comment by cferreiragonz** [comment]
@Mergifyio backport 2.12.x 2.11.x 2.10.x 2.6.x

**Comment by cferreiragonz** [comment]
@richiprosima please test this

**Comment by MiguelCompany** [comment]
@richiprosima Please test windows

**Comment by cferreiragonz** [comment]
@richiprosima please test mac

**Comment by cferreiragonz** [comment]
@richiprosima please test mac

**Comment by EduPonz** [comment]
@richiprosima please test windows test mac

**Comment by cferreiragonz** [comment]
@richiprosima please test mac

**Review by EduPonz — APPROVED** [review]

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Comment by cferreiragonz** [comment]
@richiprosima please test windows test linux check style

**Comment by EduPonz** [comment]
@richiprosima please test windows

**Review by EduPonz — APPROVED** [review]

**Review by EduPonz — APPROVED** [review]

**Pull request merged**